src/sentry/digests/utils.py

```diff
@@ -97,10 +104,23 @@ def get_event_from_groups_in_digest(digest: Digest) -> Iterable[Event]:
     }
 
 
-def build_custom_digest(original_digest: Digest, events: Iterable[Event]) -> Digest:
+def build_custom_digest(
+    original_digest: Digest, events: Iterable[Event], participant: RpcActor
+) -> Digest:
     """Given a digest and a set of events, filter the digest to only records that include the events."""
     user_digest: Digest = {}
+    rule_snoozes = RuleSnooze.objects.filter(
+        Q(Q(user_id=participant.id) | Q(user_id__isnull=True)), rule__in=original_digest.keys()
```

Comment:
Nit: Btw I realise I lied, you don't need the outer `Q` here, since it's a separate argument to filter anyway